pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -54,7 +55,45 @@ func NewController() *Controller {
 func (c *Controller) AddRegistry(registry Registry) {
 	c.storeLock.Lock()
 	defer c.storeLock.Unlock()
-	c.registries = append(c.registries, registry)
+	registries := c.registries
+	registries = append(registries, registry)
+	c.registries = registries
+}
+
+// DeleteRegistry deletes specified registry from the aggregated controller
+func (c *Controller) DeleteRegistry(registry Registry) {
+	c.storeLock.Lock()
+	defer c.storeLock.Unlock()
+	registries := c.registries
+	if len(c.registries) == 0 {
+		log.Warnf("Registry list is empty, nothing to delete")
+		return
+	}
+	index, ok := c.GetRegistryIndex(registry)
+	if !ok {
+		log.Warnf("Registry is not found in the registries list, nothing to delete")
+		return
+	}
+	registries = append(registries[:index], registries[index+1:]...)
+	c.registries = registries
```

Comment:
I guess that the caller to this method will have more things to do such as removing the controller itself, terminating the goroutine associated with this controller, etc.